py/frequenz/client/electricity_trading/_types.py

```diff
@@ -267,23 +283,39 @@ def from_pb(
 
         Returns:
             Enum value corresponding to the protobuf message.
+
+        Raises:
+            ValueError: If the duration is not 5, 15, 30, or 60 minutes.
         """
-        if not any(e.value == delivery_duration for e in cls):
-            _logger.warning(
-                "Unknown delivery duration %s. Returning UNSPECIFIED.",
-                delivery_duration,
+        mapping = {
+            delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_5: timedelta(
+                minutes=5
+            ),
+            delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_15: timedelta(
+                minutes=15
+            ),
+            delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_30: timedelta(
+                minutes=30
+            ),
+            delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_60: timedelta(
+                minutes=60
+            ),
+        }
+        if delivery_duration not in mapping:
+            raise ValueError(
+                f"Invalid protobuf delivery duration value: {delivery_duration}"
```

Comment:
this is changing the previous behaviour, IIUC this should build UNSPECIFIED enum instead